File: src/main/java/com/je/spring/rest/master/service/TrackingService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.je.spring.rest.master.service;

import com.je.spring.rest.master.model.Tracking;
import java.util.List;
/**
 *
 * @author Prasetyo
 */
public interface TrackingService {
    
    public List<Tracking> getAll();
    public Tracking getById(int id);
    public long count();
    public void insert(Tracking tracking);
    
}

File: src/main/webapp/js/app/transaction/tracking.service.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


(function() {

	var services = angular.module('je-transaction-tracking-service', [ 'ngResource' ]);

	
	services.factory('TrackingListFactory', [ '$resource', function($resource) {

		return $resource('service/transaksi/tracking', {}, {
			query : {
				method : 'GET',
				isArray : false
			},
			create : {
				method : 'POST'
			}
		});

	} ]);
	
	services.factory('TrackingEditFactory', [ '$resource', function($resource) {

		return $resource('service/transaksi/tracking/:id', {}, {
			show : {
				method : 'GET',
				params: {id: '@id'}
			},
			update : {
				method : 'PUT',
				params :{id: '@id'}
			}
		});

	} ]);
	
})();